pkgs/stdenv/generic/make-derivation.nix

```
@@ -163,6 +163,31 @@ let
 
 , env ? { }
 
+, src ? null
+
+, srcWorkDir ? null
+
+, sourceRoot ?
+    if srcWorkDir == null then
+      null
+    else if ! builtins.isPath srcWorkDir then
+      throw "stdenv.mkDerivation: The `srcWorkDir` attribute is expected to be a path, but it's a ${builtins.typeOf srcWorkDir} instead."
```

```suggestion
      throw "stdenv.mkDerivation: The `srcWorkDir` attribute must be a path, but a ${builtins.typeOf srcWorkDir} was given."
```

A bit less verbose, and more consistent with the other errors.